pyproject.toml

```diff
@@ -1,9 +1,127 @@
-[build-system]
-requires = [
-    "setuptools>=42",
-    "wheel"
+[tool.poetry]
+name = "tidy3d"
+version = "2.6.0rc1"  # Thinking about bump2version
+description = "A fast FDTD solver"
+authors = ["Tyler Hughes <tyler@flexcompute.com>"]
+license = "LGPLv2+"
+readme = "README.md"
+homepage = "https://github.com/flexcompute/tidy3d"
+repository = "https://github.com/flexcompute/tidy3d"
+classifiers = [
+    "License :: OSI Approved :: GNU Lesser General Public License v2 or later (LGPLv2+)",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Operating System :: OS Independent",
+]
+documentation = "https://docs.flexcompute.com/projects/tidy3d/en/latest/"
+
+[tool.poetry.urls]
+"Bug Tracker" = "https://github.com/flexcompute/tidy3d/issues"
+
+[tool.poetry.dependencies]
+python = ">=3.9,<3.12"
+pyroots = ">=0.5.0"
+xarray = ">=0.16.2"
+importlib-metadata = "^6.0.0"
+h5netcdf = "1.0.2"
+h5py = "^3.0.0"
+rich = "<12.6.0"
+matplotlib = "*"
+shapely = "^2.0"
+pydantic = "^2.0"
+PyYAML = "*"
+dask = "*"
+toml = "*"
+scipy = "*"
+boto3 = "1.23.1"
+requests = "*"
+pyjwt = "*"
+click = "8.0.3"
+responses = "*"
+
+### Optional dependencies ### 
+# development core
+pre-commit = {version="*", optional = true}
+black = {version="22.3.0", optional = true}
+ruff = {version="*", optional = true}
+tox = {version="*", optional = true}
+pylint = {version="*", optional = true}
+pytest = {version="*", optional = true}
+pytest-timeout = {version="*", optional = true}
+memory_profiler = {version="*", optional = true}
+dill = {version="*", optional = true}
+ipython = {version="*", optional = true}
+
+# gdspy
+gdspy = {version="*", optional = true}
+
+# gdstk
+gdstk = {version="*", optional = true}
```

Comment:
Yeah so I made a PR for that, I think Joaquin preferred it being fixed rather than restricting it as apparently it'd create some conflicts on gdsfactory of them using a new version. We can also come back to this closer to the release as maybe it's something Lucas was working on?

Reply:
sounds good.